src/ng/q.js

```diff
@@ -354,8 +354,8 @@ function qFactory(nextTick, exceptionHandler) {
       var that = this;
       var done = false;
       try {
-        if ((isObject(val) || isFunction(val))) then = val && val.then;
-        if (isFunction(then)) {
+        if(isPromiseLike(val)) {
+          then = val && val.then
```

Comment:
To avoid it being mutated in-between calls is what their point is, but that cannot reasonably happen here since this happens inside of a single function call and JS is single threaded.

The only way this would even happen is like if they had a getter that returned a different function when the property was accessed which is....nuts.